src/main/java/co/cask/gcp/gcs/source/GCSSource.java

```diff
@@ -93,89 +69,153 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     private String path;
 
+    @Macro
     @Nullable
     @Description("Map of properties to set on the InputFormat.")
-    @Macro
     private String fileSystemProperties;
 
-    @Nullable
-    @Description("Maximum split-size for each mapper in the MapReduce " +
-      "Job. Defaults to 128MB.")
     @Macro
+    @Nullable
+    @Description("Maximum size of each partition used to read data. "
+      + "Smaller partitions will increase the level of parallelism, but will require more resources and overhead.")
     private Long maxSplitSize;
 
     @Nullable
-    @Description("If specified, each output record will include a field with this name that contains the file URI " +
-      "that the record was read from. Requires a customized version of CombineFileInputFormat, so it cannot be used " +
-      "if an inputFormatClass is given.")
+    @Description("Output field to place the path of the file that the record was read from. "
+      + "If not specified, the file path will not be included in output records. "
+      + "If specified, the field must exist in the output schema as a string.")
     private String pathField;
 
+    @Macro
+    @Nullable
+    @Description("Format of the data to read. Supported formats are 'text', 'avro' or 'parquet'. "
+      + "The default value is 'text'.")
+    private String format;
+
     @Nullable
     @Description("Output schema. If a Path Field is set, it must be present in the schema as a string.")
     private String schema;
 
     @Nullable
-    @Description("If true and a pathField is specified, only the filename will be used. If false, the full " +
-      "URI will be used. Defaults to false.")
+    @Description("Whether to only use the filename instead of the URI of the file path when a path field is given. "
+      + "The default value is false.")
     private Boolean filenameOnly;
 
+    @Macro
+    @Nullable
+    @Description("Regular expression that file paths must match in order to be included in the input. "
+      + "The full file path is compared, not just the file name."
+      + "If no value is given, no file filtering will be done. "
+      + "See https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html for more information about "
+      + "the regular expression syntax.")
+    private String fileRegex;
+
     @Nullable
-    @Description("Boolean value to determine if files are to be read recursively from the path. Default is false.")
+    @Description("Whether to recursively read directories within in the input directory. The default is false.")
     private Boolean recursive;
 
+    // this is a hidden property that only exists for wrangler's parse-as-csv that uses the header as the schema
+    // when this is true and the format is text, the header will be the first record returned by every record reader
+    @Nullable
+    private Boolean copyHeader;
+
     public GCSSourceConfig() {
-      this.maxSplitSize = 128L;
+      this.maxSplitSize = 128L * 1024 * 1024;
```

Comment:
Was earlier the split size not being used in MB?

Reply:
it was, but that made it inconsistent with the file source, which uses bytes.